frontend/src/lib/components/Cards/TextCard/TextCard.tsx

```diff
@@ -0,0 +1,289 @@
+                enableFormOnSubmit
+            >
+                <Field name="body" label="">
+                    {({ value, onChange }) => (
+                        <Tabs>
+                            <Tabs.TabPane tab="Write" key="write-card" destroyInactiveTabPane={true}>
+                                <LemonTextArea
+                                    data-attr="text-card-edit-area"
+                                    autoFocus
+                                    value={value}
+                                    onChange={(newValue) => onChange(newValue)}
+                                />
+                                <div className="text-muted inline-flex items-center space-x-1">
+                                    <IconMarkdown />
+                                    <span>Basic formatting and markdown support</span>
+                                </div>
+                            </Tabs.TabPane>
+                            <Tabs.TabPane tab="Preview" key={'preview-card'}>
+                                <TextCardBody text={value} />
+                            </Tabs.TabPane>
+                        </Tabs>
+                    )}
+                </Field>
+            </Form>
+        </LemonModal>
+    )
+}
+
+interface TextCardProps extends React.HTMLAttributes<HTMLDivElement>, Resizeable {
+    dashboardId?: string | number
+    textTile: DashboardTile
+    children?: JSX.Element
+    updateColor?: (newColor: Tileable['color']) => void
+    removeFromDashboard?: () => void
+    duplicate?: () => void
+    moveToDashboard?: (dashboard: DashboardType) => void
+    /** buttons to add to the "more" menu on the card**/
+    moreButtons?: JSX.Element | null
+}
+
+function TextCardBody({ text }: { text: string }): JSX.Element {
+    return (
+        <div className={'w-full h-full block'}>
+            <Textfit mode={text?.match(/([\r\n])/gm)?.length ? 'multi' : 'single'} min={16} max={100}>
+                <div className="TextCard-Body px-2 pb-2">
+                    <ReactMarkdown>{text}</ReactMarkdown>
+                </div>
+            </Textfit>
+        </div>
+    )
+}
+
+export function TextCardInternal(
+    {
+        textTile,
+        showResizeHandles,
+        canResizeWidth,
+        children,
+        className,
+        dashboardId,
+        updateColor,
+        moreButtons,
+        removeFromDashboard,
+        duplicate,
+        moveToDashboard,
+        ...divProps
+    }: TextCardProps,
+    ref: React.Ref<HTMLDivElement>
+): JSX.Element {
+    const { push } = useActions(router)
+    const { text } = textTile
+    if (!text) {
+        throw new Error('TextCard requires text')
+    }
+
+    const { nameSortedDashboards } = useValues(dashboardsModel)
+    const otherDashboards = nameSortedDashboards.filter((dashboard) => dashboard.id !== dashboardId)
+    return (
+        <div
+            className={clsx('TextCard border rounded flex flex-col overflow-hidden', className)}
```

Comment:
These resize handles shouldn't be cut off:
<img width="211" alt="Screen Shot 2022-10-11 at 17 23 15" src="https://user-images.githubusercontent.com/4550621/195133066-3b8ffcb1-6966-48c1-9f62-1b4b8e7a664b.png">

Reply:
Nevertheless overflowing text should be scrollable vertically (horizontally we should just break the lines up)
<img width="745" alt="Screen Shot 2022-10-12 at 12 24 04" src="https://user-images.githubusercontent.com/4550621/195318722-562c32ee-6678-48fb-8be2-b8cf9109c175.png">

Reply:
Ah, hang over from single and multiline mode of React Textfit fighting each other

Reply:
![Screenshot 2022-10-12 at 11 34 02](https://user-images.githubusercontent.com/984817/195320949-bfe778ab-7bd7-4522-a7bb-24dac356c16c.png)